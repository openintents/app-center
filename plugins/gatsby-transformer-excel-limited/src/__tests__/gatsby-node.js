const Promise = require(`bluebird`)
const XLSX = require(`xlsx`)

const { onCreateNode } = require(`../gatsby-node`)

describe(`Process nodes correctly`, () => {
  const node = {
    id: `whatever`,
    parent: null,
    children: [],
    extension: `csv`,
    internal: {
      contentDigest: `whatever`,
      mediaType: `text/csv`,
    },
    name: `test`,
  }

  // Make some fake functions its expecting.
  const loadNodeContent = node => Promise.resolve(node.content)

  it(`correctly creates nodes from JSON which is an array of objects`, async () => {
    const data = [[`blue`, `funny`], [true, `yup`], [false, `nope`]]
    const csv = XLSX.utils.sheet_to_csv(XLSX.utils.aoa_to_sheet(data))
    node.content = csv

    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode({
      node,
      loadNodeContent,
      actions,
      createNodeId,
      createContentDigest,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(2 + 1)
      expect(createParentChildLink).toHaveBeenCalledTimes(2 + 1)
      expect(createContentDigest).toHaveBeenCalledTimes(2 + 1)
    })
  })

  it(`should correctly create nodes from JSON with raw option false`, async () => {
    const data = [[`red`, `veryfunny`], [true, `certainly`], [false, `nada`]]
    const csv = XLSX.utils.sheet_to_csv(XLSX.utils.aoa_to_sheet(data))
    node.content = csv

    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode(
      {
        node,
        loadNodeContent,
        actions,
        createNodeId,
        createContentDigest,
      },
      { raw: false }
    ).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(2 + 1)
      expect(createParentChildLink).toHaveBeenCalledTimes(2 + 1)
      expect(createContentDigest).toHaveBeenCalledTimes(2 + 1)
    })
  })

  it(`should correctly create nodes from JSON with legacy rawOutput option false`, async () => {
    const data = [[`red`, `veryfunny`], [true, `certainly`], [false, `nada`]]
    const csv = XLSX.utils.sheet_to_csv(XLSX.utils.aoa_to_sheet(data))
    node.content = csv

    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode(
      {
        node,
        loadNodeContent,
        actions,
        createNodeId,
        createContentDigest,
      },
      { rawOutput: false }
    ).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(2 + 1)
      expect(createParentChildLink).toHaveBeenCalledTimes(2 + 1)
      expect(createContentDigest).toHaveBeenCalledTimes(2 + 1)
    })
  })
})
