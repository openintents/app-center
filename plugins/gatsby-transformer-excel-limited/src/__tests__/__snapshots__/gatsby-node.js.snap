// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process nodes correctly correctly creates nodes from JSON which is an array of objects 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Process nodes correctly correctly creates nodes from JSON which is an array of objects 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "blue,funny
TRUE,yup
FALSE,nope
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "blue,funny
TRUE,yup
FALSE,nope
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "blue,funny
TRUE,yup
FALSE,nope
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Process nodes correctly should correctly create nodes from JSON with legacy rawOutput option false 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
      "red": "TRUE",
      "veryfunny": "certainly",
    },
  ],
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
      "red": "FALSE",
      "veryfunny": "nada",
    },
  ],
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Process nodes correctly should correctly create nodes from JSON with legacy rawOutput option false 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
        "red": "TRUE",
        "veryfunny": "certainly",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
        "red": "FALSE",
        "veryfunny": "nada",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Process nodes correctly should correctly create nodes from JSON with raw option false 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
      "red": "TRUE",
      "veryfunny": "certainly",
    },
  ],
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv__Sheet1",
      },
      "parent": "whatever",
      "red": "FALSE",
      "veryfunny": "nada",
    },
  ],
  Array [
    Object {
      "children": Array [],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "TestCsv",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Process nodes correctly should correctly create nodes from JSON with raw option false 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
        "red": "TRUE",
        "veryfunny": "certainly",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv__Sheet1",
        },
        "parent": "whatever",
        "red": "FALSE",
        "veryfunny": "nada",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "TestCsv",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "red,veryfunny
TRUE,certainly
FALSE,nada
",
        "extension": "csv",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/csv",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
]
`;
